### ISSUE TITLE
MGLMapViewDelegate is unable to block moving pinch gesture

### ISSUE THREAD

**Issue by jmkiley** [mapbox/mapbox-gl-native#9168]
I am still able to change the map camera even when [mapView:shouldChangeFromCamera:toCamera:](https://www.mapbox.com/ios-sdk/api/3.6.0-beta.2/Protocols/MGLMapViewDelegate.html#/c:objc(pl)MGLMapViewDelegate(im)mapView:shouldChangeFromCamera:toCamera:) returns false. I can pan using a pinching gesture.

Code to reproduce is based on the [restricting panning](https://www.mapbox.com/ios-sdk/examples/constraining-gestures/) example. Repro'ed with that example using Mapbox iOS SDK v3.5.4 and v3.6.0-beta.2.

```
class BlockingGesturesDelegateExample_Swift: UIViewController, MGLMapViewDelegate {
    
    private var colorado: MGLCoordinateBounds!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let mapView = MGLMapView(frame: view.bounds)
        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        mapView.delegate = self
        
        // Denver, Colorado
        let center = CLLocationCoordinate2D(latitude: 39.748947, longitude: -104.995882)
        
        // Starting point
        mapView.setCenter(center, zoomLevel: 10, direction: 0, animated: false)
        
        // Colorado's bounds
        let ne = CLLocationCoordinate2D(latitude: 40.989329, longitude: -102.062592)
        let sw = CLLocationCoordinate2D(latitude: 36.986207, longitude: -109.049896)
        colorado = MGLCoordinateBounds(sw: sw, ne: ne)
        
        view.addSubview(mapView)
 }

    func mapView(_ mapView: MGLMapView, shouldChangeFrom oldCamera: MGLMapCamera, to newCamera: MGLMapCamera) -> Bool {
     
        return false
  }
}
```

**Comment by 1ec5** [comment]
This is by design. `-mapView:shouldChangeFromCamera:toCamera:` is intended as a way to restrict camera changes due to user gestures. To prevent programmatic changes to the camera, remove the code that changes the camera. 😉

**Comment by 1ec5** [comment]
I can’t reproduce this issue on master (as of 40e47556214e999c952c1e88e026d673cdd96c72), nor on the release-ios-v3.6.0-android-v5.1.0 branch (as of d3730241bbc98ec9a44295e3d3f1e7318d4cf73e). My attempted steps were to make the change below in iosapp, then go to zoom level 18 and perform a two-finger pan gesture in the simulator.

```diff
diff --git a/platform/ios/app/MBXViewController.m b/platform/ios/app/MBXViewController.m
index 29c5c6501..9c821bd2c 100644
--- a/platform/ios/app/MBXViewController.m
+++ b/platform/ios/app/MBXViewController.m
@@ -1922,4 +1922,8 @@ typedef NS_ENUM(NSInteger, MBXSettingsMiscellaneousRows) {
     }
 }
 
+- (BOOL)mapView:(MGLMapView *)mapView shouldChangeFromCamera:(MGLMapCamera *)oldCamera toCamera:(MGLMapCamera *)newCamera {
+    return newCamera.pitch == 0;
+}
+
 @end
```

**Comment by 1ec5** [comment]
OK, I can reproduce on a device. To reproduce, pinch with two fingers. The map won’t zoom, but it will pan as the fingers move together or apart.

**Comment by 1ec5** [comment]
Fixed in #9344 on the release-ios-v3.6.0-android-v5.1.0 branch.
